hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -479,6 +477,9 @@ bool XmlChangeset::addWay(ChangesetInfoPtr& changeset, ChangesetType type, XmlWa
         }
       }
     }
+    //  Add the way last
+    changeset->add(ElementType::Way, type, way->id());
+    markBuffered(way);
```

Comment:
Changed up the ordering in the changesets so that all elements depended on get added first.